app/src/actions/search.js

```diff
@@ -1,30 +1,43 @@
-import { GET_SEARCH_RESULTS } from '../actions';
+import { SET_SEARCH_RESULTS } from '../actions';
 import 'whatwg-fetch';
 
+/**
+ * Reset the search results to their initial state
+ *
+ * @export resetSearchResults
+ * @returns {Object}
+ */
+export function resetSearchResults() {
+  return {
+    type: SET_SEARCH_RESULTS,
+    payload: {
+      entries: [],
+      count: -1
+    }
+  };
+}
+
 export function getSearchResults(searchTerm) {
   return (dispatch, getState) => {
     const state = getState();
 
     if (!state.user.token) {
```

Comment:
this is totally fine, but a slightly better way to do it is to return early when requirements fail: https://github.com/Vizzuality/GlobalFishingWatch/blob/master/app/src/actions/vesselInfo.js#L7

by doing so, the actual logic has less "if" "else" and whatever around it, making it easier to understand.